proxy/proxy.go

```diff
@@ -0,0 +1,134 @@
+// This is Trojan horse to change DNS resolution to be universally correct independent of system configuration
+
+package dns_proxy
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	rdns "github.com/folbricht/routedns"
+	"github.com/miekg/dns"
+)
+
+type Proxy struct {
+	server *rdns.DNSListener
+
+	resolutionCache map[string]*dns.Msg
+	cacheLock       sync.RWMutex
+}
+
+// Serve starts server and redirects all inputs to the target DNS
+// This function blocks until server is stopped
+func (p *Proxy) Serve(port int, network, remoteDNS string) error {
+	if p.server != nil {
+		return fmt.Errorf("server already exist")
+	}
+	p.server = &rdns.DNSListener{
+		Server: &dns.Server{
+			Addr:    fmt.Sprintf(":%d", port),
```

Comment:
why do we set only port here?

Reply:
because we're serving on the localhost only